syntax = "proto3";

package imgTrans;

service Communicate {
    rpc registerImgTransService(RegisterImgTransServiceRequest) returns (RegisterImgTransServiceResponse) {}
    rpc unregisterImgTransService(UnregisterImgTransServiceRequest) returns (UnregisterImgTransServiceResponse) {}
    rpc getImg(GetImgRequest) returns (GetImgResponse) {}
}

message NullRequest {}
message NullResponse {}

message UnregisterImgTransServiceRequest {
    int64 connectId = 1;
}

message UnregisterImgTransServiceResponse {
}

message RegisterImgTransServiceRequest {
    message Argument {
        string key = 1;
        string value = 2;
    }
    string imgType = 1;
    repeated Argument args = 2;
}

message RegisterImgTransServiceResponse {
    int64 connectId = 1;
}

message GetImgRequest {
    int64 connectId = 1;
}

message GetImgResponse {
    int32 w = 1;
    int32 h = 2;
    int64 imgId = 3;
    bytes buf = 4;
}

// protoc --grpc_out=. --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` img_trans.proto
// protoc --cpp_out=. ./img_trans.proto 